docs/blog/posts/2023/sr-linux-kubernetes-anycast-lab.md

```diff
@@ -0,0 +1,955 @@
+2. `bash /hostname.sh` is just a hacky script that updates the shell with the name of the container, so it's easier to see where we are connected when we attach to the client containers.
+
+Containerlab will configure the Leaf/Spine fabric at boot time. You can check the configurations in the [config][clab-configs] folder.
+
+Minikube nodes are referenced by using the `kind: ext-container` option. This option instructs Containerlab to wait for containers with the name declared (`cluster1` in the example) to appear. The moment they appear, clab will take care of creating the interface (`192.168.1.11/24 dev eth1`) and default route. Take note that Minikube node creation it's not managed by Containerlab, is directly managed by the Minikube client.
+
+Later in the blog post we will carefully explain the process to fully boot up the Lab.
+
+!!!tip
+    Consult with [containerlab](https://containerlab.dev/manual/kinds/ext-container/) documentation to learn more about the `ext-container` kind.
+
+
+### links
+And finally, let's see how we interconnect all elements:
+
+```yaml title="Defining links"
+topology:
+# -- snip --
+
+  links:
+  ### #### fabric ### ####
+    - endpoints: ["spine1:e1-1", "leaf1:e1-49"]
+    - endpoints: ["spine1:e1-2", "leaf2:e1-49"]
+    - endpoints: ["spine1:e1-3", "leaf3:e1-49"]
+    - endpoints: ["spine1:e1-4", "leaf4:e1-49"]
+    - endpoints: ["spine2:e1-1", "leaf1:e1-50"]
+    - endpoints: ["spine2:e1-2", "leaf2:e1-50"]
+    - endpoints: ["spine2:e1-3", "leaf3:e1-50"]
+    - endpoints: ["spine2:e1-4", "leaf4:e1-50"]
+
+    #### minikube ####
+    - endpoints: ["leaf1:e1-1", "cluster1:eth1"]
+    - endpoints: ["leaf2:e1-1", "cluster1-m02:eth1"]
+    - endpoints: ["leaf3:e1-1", "cluster1-m03:eth1"]
+
+
+    #### clients ####
+    - endpoints: ["cli1:eth1", "leaf1:e1-2"]
+    - endpoints: ["cli2:eth1", "leaf2:e1-2"]
+    - endpoints: ["cli3:eth1", "leaf3:e1-2"]
+    - endpoints: ["cli4:eth1", "leaf4:e1-2"]
+```
+This is the full definition of all the connections required. As you can see, we use the first port (`e1-1`) of every leaf switch to connect the k8s node and second port (`e1-2`) to connect a client for tests.
+
+
+!!! note
+    `eth0`minikube node interfaces are connected to the docker bridge of the host running the topology. `eth1` interfaces, connected to Leaf switches, will be the ones used by MetalLB to establish the BGP the sessions. As reachability is signaled through this interface, clients will also reach k8s cluster services though `eth1`.
+
+
+
+## Lab deployment
+
+
+First, clone the lab:
+
+```bash
+https://github.com/srl-labs/srl-k8s-anycast-lab.git && cd srl-k8s-anycast-lab
+```
+
+Open two different shell sessions. One will be used to deploy the Containerlab topology, the other to start the Minikube cluster:
```

Comment:
do you really need to shells? Can you deploy minikube first and then proceed with clab deployment?

Reply:
no, clab complains if it sees minikube containers booted. At least that's my experience

Reply:
This shouldn't be the case. Can you show the error you saw?
/cc @steiler

Reply:
WARN[0000] Failed to decode the keys ["storage.options.override_kernel_check"] from "/etc/containers/storage.conf"
INFO[0000] Containerlab v0.42.0 started
INFO[0000] Parsing & checking topology file: srl-k8s-lab.clab.yml
INFO[0000] Nokia SR Linux v23.3.1+ requires a kernel version greater than 4.10.0. Detected kernel version: 5.4.240
INFO[0000] Nokia SR Linux v23.3.1+ requires a kernel version greater than 4.10.0. Detected kernel version: 5.4.240
INFO[0000] Nokia SR Linux v23.3.1+ requires a kernel version greater than 4.10.0. Detected kernel version: 5.4.240
INFO[0000] Nokia SR Linux v23.3.1+ requires a kernel version greater than 4.10.0. Detected kernel version: 5.4.240
INFO[0000] Nokia SR Linux v23.3.1+ requires a kernel version greater than 4.10.0. Detected kernel version: 5.4.240
INFO[0000] Nokia SR Linux v23.3.1+ requires a kernel version greater than 4.10.0. Detected kernel version: 5.4.240
Error: containers ["cluster1-m02" "cluster1-m03" "cluster1"] already exist. Add '--reconfigure' flag to the deploy command to first remove the containers and then deploy the lab

Reply:
Unfortunately the reported behaviour reflects the actual state.
I need to fix that.

Reply:
for a description of the root cause see https://github.com/srl-labs/containerlab/issues/1513

Reply:
I have pulled https://github.com/srl-labs/containerlab/commit/fbf59f948e92609d9c405020823f57fb4af0ef9a
and while it fixes ext-container issue, now the lab is not working any more. What I can see is that the BGP session between leaf switches and metallb speakers is not established. From network pov everything looks ok (ping, arp..) but tcp bgp session is not established. I see other big commit related to linking, maybe some change there is breaking this type of connectivity. I will troubleshoot further.

Reply:
we released 0.44.0 and I upgraded the table to mention it, but keep us posted if you see what happened.
I don't think links PR is at fault here

Reply:
I just did a quick test with https://containerlab.dev/lab-examples/srl-frr/ and I'm getting same result, the bgp session between srl and frr is not working:

# ip netns exec clab-srlfrr01-srl tcpdump -nni e1-1-0 tcp port  179
tcpdump: verbose output suppressed, use -v or -vv for full protocol decode
listening on e1-1-0, link-type EN10MB (Ethernet), capture size 262144 bytes
02:51:10.920467 IP 192.168.1.2.49476 > 192.168.1.1.179: Flags [S], seq 4266912591, win 64240, options [mss 1460,sackOK,TS val 202403849 ecr 0,nop,wscale 7], length 0
02:51:12.968125 IP 192.168.1.2.49476 > 192.168.1.1.179: Flags [S], seq 4266912591, win 64240, options [mss 1460,sackOK,TS val 202405897 ecr 0,nop,wscale 7], length 0

srl displays:
Group ibgp: Peer 192.168.1.2 moved from higher state ACTIVE to lower state IDLE due to event TCP SOCKET ERROR

not sure what's happening but it's not just related to ext-container

Reply:
By downgrading to 0.34 it fixes itself?

Reply:
Take 0.44.1 
It should contain a fix which was disabling tx offload for veth interfaces

Reply:
Fixed!! I test it with 0.44.1 and now it's working. Thanks for the quick fix!

Reply:
changed clab release in lab description